src/prototype_room_init.js

```diff
@@ -97,11 +133,6 @@ function setStructures(room, path, costMatrixBase) {
         costMatrixBase.set(structurePos.x, structurePos.y, config.layout.structureAvoid);
         continue;
       }
-      if (room.memory.position.structure.lab.length < CONTROLLER_STRUCTURES.lab[8]) {
-        room.memory.position.structure.lab.push(structurePos);
-        costMatrixBase.set(structurePos.x, structurePos.y, config.layout.structureAvoid);
-        continue;
-      }
```

Comment:
If the path is not long enough if will be ignored. As seen in current rooms where the labs are missing.